src/map/status.cpp

```diff
@@ -12247,6 +12341,16 @@ int status_change_end_(struct block_list* bl, enum sc_type type, int tid, const
 			}
 			break;
 
+		case SC_SOULUNITY:
+			{
+				struct block_list *d_bl = map_id2bl(sce->val1);
+				if( d_bl ) {
+					if( d_bl->type == BL_PC )
```

Comment:
You could move this into one check:
```suggestion
					if( d_bl && d_bl->type == BL_PC )
```

Reply:
done.